openapi_python_client/templates/model.pyi

```diff
@@ -24,11 +24,17 @@ class {{ model.reference.class_name }}:
         {% endif %}
         {% endfor %}
 
-        return {
-            {% for property in model.required_properties + model.optional_properties %}
-            "{{ property.name }}": {{ property.python_name }},
-            {% endfor %}
-        }
+        properties: Dict[str, Any] = dict()
+
+        {% for property in model.required_properties + model.optional_properties %}
+        {% if not property.required %}
+        if {{property.python_name}} is not UNSET:
+            properties["{{ property.name }}"] = {{ property.python_name }}
+        {% else %}
+        properties["{{ property.name }}"] = {{ property.python_name }}
```

Comment:
Is it possible that a property is both required AND unset, and if so is it desired that it would be exposed in the dictionary directly (instead of, say, erroring)?

Reply:
If a property is required, we shouldn't allow unset - we simply shouldn't allow a default. It could still be nullable and required, in which case we should allow it to be `None`.